### ISSUE TITLE
Mysterious number 1 on MULTI_STEP pages

### ISSUE THREAD

**Issue by laurensvalk** [trevorsandy/lpub3d#449]
What is the number `1` on the top left of this page?

I have multiple pages with multi-steps (though the example below has only one). The number in the top left is always `1`, so it is not a page number. I can't seem to remove it, either. I tried to workaround it by making it white like the background, but that also makes the correct step numbers `1`, `2` invisible.

Due to the presence of this `1`, the PLI is also not in the same position as on pages with regular steps.

![image](https://user-images.githubusercontent.com/12326241/94306910-16153800-ff74-11ea-9ce6-7c63128a2722.png)

Minimal example:

```
0 FILE main.ldr
0 main
0 Name: main.ldr
0 Author: MLCad
0 Unofficial Model
0 ROTATION CENTER 0 0 0 1 "Custom"
0 ROTATION CONFIG 0 0
0 !LPUB CALLOUT PLI PER_STEP GLOBAL FALSE
0 !LPUB MULTI_STEP PLI PER_STEP FALSE
0 !LPUB PAGE DISPLAY_PAGE_NUMBER GLOBAL FALSE
0 !LPUB PLI PLACEMENT GLOBAL TOP_LEFT PAGE INSIDE
0 !LPUB PAGE BACKGROUND GLOBAL COLOR "#ffffff"
0 !LPUB PAGE ORIENTATION GLOBAL LANDSCAPE
0 !LPUB STEP_NUMBER PLACEMENT GLOBAL BOTTOM LEFT PLI OUTSIDE
1 0 0 0 0 1 0 0 0 1 0 0 0 1 right_motors_rear.ldr
0 ROTSTEP 30 135 0 ABS
0 NOFILE 
0 FILE right_motors_rear.ldr
0 New Model
0 Name: right_motors_rear.ldr
0 Author: MLCad
0 Unofficial Model
0 !LPUB MULTI_STEP BEGIN
0 !LPUB MULTI_STEP ALLOC HORIZONTAL
0 ROTATION CENTER 0 0 0 1 "Custom"
0 ROTATION CONFIG 0 0
1 0 20 -240 120 1 0 0 0 -1 0 0 0 -1 64179.dat
1 71 10 -240 40 0 -1 0 0 0 1 -1 0 0 32138.dat
1 71 60 -240 160 1 0 0 0 1 0 0 0 1 32054.dat
0 ROTSTEP 30 45 0 ABS
1 71 40 -240 100 0 0 1 -1 0 0 0 -1 0 48989.dat
0 !LPUB CALLOUT BEGIN
1 1 30 -240 160 1 0 0 0 -1 0 0 0 -1 axlepinf.ldr
0 !LPUB CALLOUT POINTER BOTTOM 0.377 0.694 0.212 20.000 20.000 -52.180 -74.356 -52.000 0.000 0.000 1
0 !LPUB CALLOUT POINTER_ATTRIBUTE BORDER 1 #ffffff 0.015 1
0 !LPUB CALLOUT PLACEMENT TOP CENTER ASSEM OUTSIDE OFFSET 0.2359 -0.3433
0 !LPUB CALLOUT END
0 STEP
0 !LPUB MULTI_STEP END
0
0 FILE axlepinf.ldr
0 New Model
0 Name: axlepinf.ldr
0 Author: MLCad
0 Unofficial Model
0 ROTATION CENTER 0 0 0 1 "Custom"
0 ROTATION CONFIG 0 0
1 1 0 0 0 1 0 0 0 1 0 0 0 1 3749.dat
0
0
0 NOFILE 
```

**Comment by trevorsandy** [comment]
This behaviour is as designed. I may take a look at making the step number optional which is to say add a setting to, optionally, not display the 'step group' number when PLI per step  is set to off on a multi-step page. Take a look at #382 and #388.

Cheers,

**Comment by laurensvalk** [comment]
> This behaviour is as designed.

I think I was suprised that it was *always* `1` in my use case, so the number didn't seem to make sense in my use case. I essentially have a few pages like the page above until you are done building.

So unlike the #382 and #388 where it says `3`, it says `1` on all my 10 pages of multi-steps.

**Comment by trevorsandy** [comment]
Very well. Could you share an example model file that produces the behaviour you are describing. I'm taking another look at the designed behavior. I noticed I did not expose any controls in the GUI allowing the user to set wheter or not the step number should be displayed when PLI per step is set to FALSE on a multi-step page. 

However, the meta command is enabled. It is 0 !LPUB ASSEM SHOW_STEP_NUMBER FALSE. That's the good news! The bad news is I forgot to initialize the PLI data object when the step number is not shown so there is no PLI information available for the placement routines to act on; hence, it is placed arbitrarily at the top left 'outside' corner of the page.. This obviously is a bug, which I have already corrected in soon to be released 2.4.0. As you can see I've reopned this ticket.

If you can send an example where by the step group step number is not 1 but is showing such I will take a look.

Cheers,

**Comment by laurensvalk** [comment]
Thanks for the quick response, I appreciate it.

[pyramid.mpd.txt](https://github.com/trevorsandy/lpub3d/files/5286598/pyramid.mpd.txt)

Here's an example in pyramid. If you imagine this for a much taller pyramid, you can see why the number 1 on the top left may not be the most suited.

![image](https://user-images.githubusercontent.com/12326241/94339845-67b5d500-fffd-11ea-8f2d-aaa1dd962f5c.png)

![image](https://user-images.githubusercontent.com/12326241/94339847-6d131f80-fffd-11ea-88b7-29edec52faa8.png)

**Comment by laurensvalk** [comment]
But for context, the basic use case I am after is the following (I made this in LPUB4 in 2013).

So in essense, just regular steps numbered incrementally. I just use multi steps to fit several steps on a page.

[![image](https://user-images.githubusercontent.com/12326241/94340016-b617a380-fffe-11ea-91da-d841bd6c201c.png)](https://www.lego.com/cdn/cs/set/assets/blt8d8677b8321b803e/RAC3_TRUCK.pdf)

**Comment by trevorsandy** [comment]
Very well. Thanks for the additional information.

I added an enhancement to 2.4.0 that allows the editor to turn hide the step number and a further flag to count the number when more than one step group exists in the submodel.

![Screenshot - 27_09_2020 , 13_42_46](https://user-images.githubusercontent.com/13388970/94364169-b5970f80-00c7-11eb-8421-cd4bdfc83d47.png)

The original enhancement in #382 was designed for a single page display where all steps in the submodel would display on a single page. Like this, the group step number would increment from the previous page and on to the next page. However, your scenario presents multiple step groups within a submodel so I've added the 'count' flag to allow the user to continuously count across pages generated by the submodel:

![Screenshot - 27_09_2020 , 13_55_51](https://user-images.githubusercontent.com/13388970/94364457-9bf6c780-00c9-11eb-8cdb-c1f846bc6cb0.png)

Cheers,
